crates/worker/src/partition_processor_manager/processor_state.rs

```diff
@@ -347,17 +364,13 @@ impl StartedProcessor {
         self.watch_rx.borrow().last_observed_leader_epoch
     }
 
-    pub fn step_down(&self) -> Result<(), TrySendError<PartitionProcessorControlCommand>> {
-        self.control_tx
-            .try_send(PartitionProcessorControlCommand::StepDown)
+    pub fn step_down(&self) {
+        self.control_tx.send_replace(TargetLeaderState::Follower);
     }
 
-    pub fn run_for_leader(
-        &self,
-        leader_epoch: LeaderEpoch,
-    ) -> Result<(), TrySendError<PartitionProcessorControlCommand>> {
+    pub fn run_for_leader(&self, leader_epoch: LeaderEpoch) {
         self.control_tx
-            .try_send(PartitionProcessorControlCommand::RunForLeader(leader_epoch))
+            .send_replace(TargetLeaderState::Leader(leader_epoch));
```

Comment:
Are we sure that we always call `run_for_leader` with value different from what's in the channel? if not, then send_if_modified might avoid unnecessary awakes on no-change scenario.

Reply:
At the moment, this should be the case. Changing it should allow us to more easily change the users in the future if necessary.